packages/react-native-reanimated/src/layoutReanimation/animationBuilder/commonTypes.ts

```diff
@@ -32,6 +32,29 @@ export interface KeyframeProps extends StyleProps {
   easing?: EasingFunction;
 }
 
+type FirstFrame =
+  | {
+      0: KeyframeProps & { easing?: never };
+      from?: never;
+    }
+  | {
+      0?: never;
+      from: KeyframeProps & { easing?: never };
+    };
+
+type LastFrame =
+  | { 100?: KeyframeProps; to?: never }
+  | { 100?: never; to: KeyframeProps };
+
+export type ValidKeyframeProps = FirstFrame &
+  LastFrame &
+  Record<number, KeyframeProps>;
```

Comment:
Unfortunately such intersections cause type shadowing:

<img width="908" alt="Screenshot 2024-08-23 at 16 01 08" src="https://github.com/user-attachments/assets/d032b409-b587-4ffa-8cd4-ffa4772314ea">

Reply:
As we have discussed async, there was a typescript issue that is no longer present in version we use in our workspace and the problem no longer exists 🥳